examples/regression/2-advanced-analysis/plot_conditional_coverage.py

```diff
@@ -0,0 +1,322 @@
+
+    return X, y
+
+
+# Data generation
+min_x, max_x, n_samples = 0, 10, 3000
+X_train, y_train = sin_with_controlled_noise(min_x, max_x, n_samples)
+X_test, y_test = sin_with_controlled_noise(min_x, max_x,
+                                           int(n_samples * split_size))
+
+# Definition of our base models
+model = LGBMRegressor(random_state=random_state, alpha=0.5)
+model_quant = LGBMRegressor(
+            objective="quantile",
+            alpha=0.5,
+            random_state=random_state
+)
+
+
+# Definition of the experimental set up
+STRATEGIES = {
+    "CV+": {
+        "method": "plus",
+        "cv": 10,
+    },
+    "JK+ab_Gamma": {
+        "method": "plus",
+        "cv": Subsample(n_resamplings=100),
+        "conformity_score": GammaConformityScore()
+    },
+    "CRF": {
+        "cv": "split",
+        "conformity_score": ConformalResidualFittingScore(
+            residual_estimator=LGBMRegressor(
+                alpha=0.5,
+                random_state=random_state),
+            split_size=0.7,
+            random_state=random_state
+        )
+    },
+    "CQR": {
+        "method": "quantile", "cv": "split", "alpha": alpha
+    },
+}
+
+y_pred, intervals, coverage, cond_coverage, coef_corr = {}, {}, {}, {}, {}
+num_bins = 10
+for strategy, params in STRATEGIES.items():
+    # computing predictions
+    if strategy == "CQR":
+        mapie = MapieQuantileRegressor(
+            model_quant,
+            **params
+        )
+        mapie.fit(X_train, y_train, random_state=random_state)
+        y_pred[strategy], intervals[strategy] = mapie.predict(X_test)
+    else:
+        mapie = MapieRegressor(model, **params, random_state=random_state)
+        mapie.fit(X_train, y_train)
+        y_pred[strategy], intervals[strategy] = mapie.predict(
+            X_test, alpha=alpha
+        )
+
+    # computing metrics
+    coverage[strategy] = regression_coverage_score_v2(
+        y_test, intervals[strategy]
+    )
+    cond_coverage[strategy] = regression_ssc_score(
+        y_test, intervals[strategy], num_bins=num_bins
+    )
+    coef_corr[strategy] = hsic(y_test, intervals[strategy])
+
+
+# Visualisation of the estimated conditional coverage
+estimated_cond_cov = pd.DataFrame(
+    columns=["global coverage", "max coverage violation", "hsic"],
+    index=STRATEGIES.keys())
+for m, cov, ssc, coef in zip(
+    STRATEGIES.keys(),
+    coverage.values(),
+    cond_coverage.values(),
+    coef_corr.values()
+):
+    estimated_cond_cov.loc[m] = [
+        round(cov[0], 2), round(ssc[0], 2), round(coef[0], 2)
+    ]
+
+with pd.option_context('display.max_rows', None, 'display.max_columns', None):
+    print(estimated_cond_cov)
+
+##############################################################################
+# We can see here that the global coverage is approximately the same for
+# all methods. What we want to understand is : "Are these methods good
+# adaptive conformal methods ?". For this we have the two metrics
```

Comment:
Maybe a bullet point list for each of the metrics would make for an easier read?

Reply:
Done